src/json-patch-duplex.ts

```diff
@@ -126,15 +133,17 @@ module jsonpatch {
   var arrOps = {
     add: function (arr, i) {
       arr.splice(i, 0, this.value);
-      return true;
+      // this may be needed when using '-' in an array
+      return {addedAt: i};
     },
     remove: function (arr, i) {
-      arr.splice(i, 1);
-      return true;
+      var removedList = arr.splice(i, 1);
```

Comment:
can't we write 
```js
return arr.splice(i, 1)[0];
```
or maybe return entire array,  to be "compliant" with `splice`